src/viewsHistory/viewsHistoryEpics.js

```diff
@@ -0,0 +1,70 @@
+import {Observable} from 'rxjs';
+import {combineEpics} from 'redux-observable';
+
+import {
+  get,
+  parseXHRError
+} from './../api/index';
+
+import {
+  UPDATE,
+} from './viewsHistoryActionTypes.js';
+import {
+  updateFulfilled,
+  updateFailure,
+} from './viewsHistoryActions.js';
+
+export const updateViewsHistory = (action$, store) => action$.ofType(UPDATE)
+  .mergeMap(({data: elements}) => {
+    const state = store.getState();
+    const headers = {
+      'Content-Type': 'application/json',
+      'X-Auth-Token': state.authReducer.tokenId,
+    };
+
+    if (!elements) {
+      return Observable.of(updateFulfilled([]));
+    }
+
+    const elementsWithoutTitle = elements
+      .map((element, index) => ({...element, index}))
+      .filter(element => !element.title);
+
+    if (elementsWithoutTitle.length === 0) {
+      return Observable.of(updateFulfilled(elements));
+    }
+
+    return Observable.zip(
+      ...elementsWithoutTitle.map(element => {
+        return get(element.url.replace('/#', state.configReducer.gohan.url), headers);
```

Comment:
Please add query for all request to download only id and name
e. g.
`_fields=id&_fields=name `